hw3/3-5/src/pages/Houses.js

```diff
@@ -0,0 +1,155 @@
+import React, { useState, useEffect } from "react";
+import Container from "react-bootstrap/Container";
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
+import "chart.js/auto";
+import { Doughnut } from "react-chartjs-2";
+
+const backgroundColors = [
+  "rgba(54, 162, 235, 0.8)",
+  "rgba(255, 206, 86, 0.8)",
+  "rgba(255, 99, 132, 0.8)",
+  "rgba(75, 192, 192, 0.8)",
+  "rgba(153, 102, 255, 0.8)",
+  "rgba(255, 159, 64, 0.8)",
+  "rgba(199, 199, 199, 0.8)",
+  "rgba(83, 102, 255, 0.8)",
+  "rgba(40, 159, 64, 0.8)",
+  "rgba(210, 199, 199, 0.8)",
+  "rgba(78, 52, 199, 0.8)",
+];
+
+const borderColors = [
+  "rgba(54, 162, 235, 1)",
+  "rgba(255, 206, 86, 1)",
+  "rgba(255, 99, 132, 1)",
+  "rgba(75, 192, 192, 1)",
+  "rgba(153, 102, 255, 1)",
+  "rgba(255, 159, 64, 1)",
+  "rgba(159, 159, 159, 1)",
+  "rgba(83, 102, 255, 1)",
+  "rgba(40, 159, 64, 1)",
+  "rgba(210, 199, 199, 1)",
+  "rgba(78, 52, 199, 1)",
+];
+
+const Validator = {
+  Targaryen: ["Targaryan", "Targaryen"],
+  Tarly: ["Tarly", "Tarth"],
+  Tyrell: ["Tyrell"],
+  Stark: ["Stark"],
+  Baratheon: ["Baratheon"],
+  Lannister: ["Lannister", "Lanister"],
+  Greyjoy: ["Greyjoy"],
+  Clegane: ["Clegane"],
+  Baelish: ["Baelish"],
+  Seaworth: ["Seaworth"],
+  Unknown: ["Unknown", "None", "Unkown"],
+  "Free Folk": ["Free Folk"],
+  Naathi: ["Naathi"],
+  Bolton: ["Bolton"],
+  Naharis: ["Naharis"],
+  Lorathi: ["Lorathi", "Lorath"],
+  Mormont: ["Mormont"],
+  Sparrow: ["Sparrow"],
+  Viper: ["Viper"],
+  Sand: ["Sand"],
+  Worm: ["Worm"],
+  Qyburn: ["Qyburn"],
+  Bronn: ["Bronn"],
+};
+
+const Thronesurl = "https://thronesapi.com/api/v2/Characters";
+
+const Houses = () => {
+  const [familyLabels, setFamilyLabels] = useState([]);
+  const [familyCounts, setFamilyCounts] = useState([]);
+  const fetchThrones = async () => {
+    try {
+      const response = await fetch(Thronesurl);
+      const data = await response.json();
+      return data;
+    } catch (error) {
+      console.error("Error:", error);
+      return [];
+    }
+  };
+
+  const countCharactersByFamily = (characters) => {
+    const familyCounts = {};
+    characters.forEach((character) => {
+      const family = character.lastName ? character.lastName.trim() : "Unknown";
+      for (const key in Validator) {
+        if (Validator[key].includes(family)) {
+          familyCounts[key] = (familyCounts[key] || 0) + 1;
+          return;
+        }
+      }
+      familyCounts[family] = (familyCounts[family] || 0) + 1;
+    });
+    return familyCounts;
+  };
+
+  const categorizeFamilyCounts = (familyCounts) => {
+    const categorizedData = {};
+    for (const key in familyCounts) {
+      if (Object.prototype.hasOwnProperty.call(familyCounts, key)) {
```

Comment:
I think we can simplify this to `if (key in familyCounts)` but... I think the key is already in the `familyCounts` as that's how we use the for loop, so this is not necessary ❓

Reply:
maybe I miss something ❓

Reply:
It basically helps in sorting.